tests/src/nonLinearElasticityTest.hh

```diff
@@ -148,3 +149,115 @@ auto NonLinearElasticityLoadControlNRandTR(const Material& mat) {
+    return denseAssembler.getMatrix(req);
+  };
+
+  auto KFunctionLE = [&](auto&& disp, auto&& lambdaLocal) -> auto& {
+    req.insertGlobalSolution(Ikarus::FESolutions::displacement, disp)
+        .insertParameter(Ikarus::FEParameter::loadfactor, lambdaLocal);
+    return denseAssemblerLE.getMatrix(req);
+  };
+
+  auto K   = KFunction(d, lambda);
+  auto KLE = KFunctionLE(d, lambda);
+  t.check(K.isApprox(KLE, tol),
+          "Mismatch between linear and non-linear stiffness matrix for zero displacements with gridDim = "
+              + std::to_string(gridDim));
+  return t;
+}
+
+template <typename Material, int gridDim = 2>
+auto SingleElementTest(const Material& mat) {
+  static_assert(gridDim == 2, "Single element test is applicable only for the 2D case");
+  TestSuite t("Single element test for non-linear Q1 element");
+
+  auto grid     = createUGGridFromCorners<gridDim>(false, true);
+  auto gridView = grid->leafGridView();
+
+  using namespace Ikarus;
+  const double tol = 1e-10;
+
+  using namespace Dune::Functions::BasisFactory;
+  auto basis = Ikarus::makeBasis(gridView, power<gridDim>(lagrange<1>()));
+
+  std::vector<Ikarus::NonLinearElasticityFE<decltype(basis), decltype(mat)>> fes;
+  for (auto& element : elements(gridView)) {
+    fes.emplace_back(basis, element, mat, nullptr, nullptr, nullptr);
+  }
+
+  auto basisP = std::make_shared<const decltype(basis)>(basis);
+  Ikarus::DirichletValues dirichletValues(basisP->flat());
+
+  auto denseAssembler = Ikarus::DenseFlatAssembler(fes, dirichletValues);
+
+  Eigen::VectorXd d;
+  d.setZero(basis.flat().size());
+  double lambda = 0.0;
+
+  d << 0, 0, 0, 0, 0, 0, 3, 3;
+
+  auto req = Ikarus::FErequirements().addAffordance(Ikarus::AffordanceCollections::elastoStatics);
+
+  auto KFunction = [&](auto&& disp, auto&& lambdaLocal) -> auto& {
+    req.insertGlobalSolution(Ikarus::FESolutions::displacement, disp)
+        .insertParameter(Ikarus::FEParameter::loadfactor, lambdaLocal);
+    return denseAssembler.getMatrix(req);
+  };
+  auto K = KFunction(d, lambda);
+
+  Eigen::SelfAdjointEigenSolver<Eigen::MatrixXd> essaK;
+  essaK.compute(K);
+  auto eigenValuesComputed = essaK.eigenvalues();
+
+  /// The eigen values are applicable only if 2x2 Gauss integration points are used
+  Eigen::VectorXd eigenValuesExpected;
+  eigenValuesExpected.setZero(basis.flat().size());
+  eigenValuesExpected << 1e-12, 1e-13, 895.29517303346217, 2591.79000211670471, 5541.79606160779595,
+      11686.52989116331387, 17028.95283090407248, 41332.88416951872696;
+  for (long i = 0; i < basis.flat().size(); ++i) {
+    if (abs(eigenValuesComputed[i]) > 1e-8) {  // Skips check for zero eigen values
+      t.check(Dune::FloatCmp::eq(abs(eigenValuesComputed[i]), eigenValuesExpected[i], tol),
+              "Mismatch in the " + std::to_string(i + 1)
+                  + "-th eigen value in single element test for four node non-linear 2D element");
```

Comment:
You can also distort the elemnt more with a more complicated d and then check that there are still two zero eigenvalues. The translational rigid body modes persit but the rotational gets a stiffness.

Reply:
You can also check with NeoHooke